src/coreclr/jit/utils.cpp

```diff
@@ -2211,6 +2211,166 @@ bool FloatingPointUtils::hasPreciseReciprocal(float x)
     return mantissa == 0 && exponent != 0 && exponent != 127;
 }
 
+//------------------------------------------------------------------------
+// isNegative: Determines whether the specified value is negative
+//
+// Arguments:
+//    val - value to check
+//
+// Return Value:
+//    True if val is negative
+//
+
+bool FloatingPointUtils::isNegative(float val)
+{
+    return *reinterpret_cast<INT32*>(&val) < 0;
+}
+
+//------------------------------------------------------------------------
+// isNegative: Determines whether the specified value is negative
+//
+// Arguments:
+//    val - value to check
+//
+// Return Value:
+//    True if val is negative
+//
+
+bool FloatingPointUtils::isNegative(double val)
+{
+    return *reinterpret_cast<INT64*>(&val) < 0;
+}
+
+//------------------------------------------------------------------------
+// isNaN: Determines whether the specified value is NaN
+//
+// Arguments:
+//    val - value to check for NaN
+//
+// Return Value:
+//    True if val is NaN
+//
+
+bool FloatingPointUtils::isNaN(float val)
+{
+    return val != val;
```

Comment:
Just wanted to call out, and our existing tests should catch if its the case; but if this file is ever compiled with `fast math`, the compiler will convert this to `return false` and that would be problematic.

Reply:
Doing a bitmask check would be "more expensive" but also "safer"